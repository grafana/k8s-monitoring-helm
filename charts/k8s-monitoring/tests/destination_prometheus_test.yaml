# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Destinations - Prometheus
templates:
  - alloy-config.yaml
tests:
  - it: creates the Alloy components for an Prometheus destination
    set:
      cluster: {name: test-cluster}
      alloy-singleton: {enabled: true, extraConfig: " ", includeDestinations: ["test"]}
      selfReporting: {enabled: false}
      destinations:
        - name: test
          type: prometheus
          url: https://prometheus.example.com
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]

  - it: allows you to disable retry on HTTP 429
    set:
      cluster: {name: test-cluster}
      alloy-singleton: {enabled: true, extraConfig: " ", includeDestinations: ["test"]}
      selfReporting: {enabled: false}
      destinations:
        - name: test
          type: prometheus
          url: https://prometheus.example.com
          queueConfig:
            retryOnHttp429: false
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]
