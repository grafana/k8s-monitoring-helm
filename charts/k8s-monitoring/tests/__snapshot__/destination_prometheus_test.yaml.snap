allows you to disable retry on HTTP 429:
  1: |
    |
      // Destination: test (prometheus)
      otelcol.exporter.prometheus "test" {
        add_metric_suffixes = true
        resource_to_telemetry_conversion = false
        forward_to = [prometheus.remote_write.test.receiver]
      }

      prometheus.remote_write "test" {
        endpoint {
          url = "https://prometheus.example.com"
          headers = {
          }
          tls_config {
            insecure_skip_verify = false
          }
          send_native_histograms = false

          queue_config {
            capacity = 10000
            min_shards = 1
            max_shards = 50
            max_samples_per_send = 2000
            batch_send_deadline = "5s"
            min_backoff = "30ms"
            max_backoff = "5s"
            retry_on_http_429 = false
            sample_age_limit = "0s"
          }

          write_relabel_config {
            source_labels = ["cluster"]
            regex = ""
            replacement = "test-cluster"
            target_label = "cluster"
          }
          write_relabel_config {
            source_labels = ["k8s_cluster_name"]
            regex = ""
            replacement = "test-cluster"
            target_label = "k8s_cluster_name"
          }
        }

        wal {
          truncate_frequency = "2h"
          min_keepalive_time = "5m"
          max_keepalive_time = "8h"
        }
      }
creates the Alloy components for an Prometheus destination:
  1: |
    |
      // Destination: test (prometheus)
      otelcol.exporter.prometheus "test" {
        add_metric_suffixes = true
        resource_to_telemetry_conversion = false
        forward_to = [prometheus.remote_write.test.receiver]
      }

      prometheus.remote_write "test" {
        endpoint {
          url = "https://prometheus.example.com"
          headers = {
          }
          tls_config {
            insecure_skip_verify = false
          }
          send_native_histograms = false

          queue_config {
            capacity = 10000
            min_shards = 1
            max_shards = 50
            max_samples_per_send = 2000
            batch_send_deadline = "5s"
            min_backoff = "30ms"
            max_backoff = "5s"
            retry_on_http_429 = true
            sample_age_limit = "0s"
          }

          write_relabel_config {
            source_labels = ["cluster"]
            regex = ""
            replacement = "test-cluster"
            target_label = "cluster"
          }
          write_relabel_config {
            source_labels = ["k8s_cluster_name"]
            regex = ""
            replacement = "test-cluster"
            target_label = "k8s_cluster_name"
          }
        }

        wal {
          truncate_frequency = "2h"
          min_keepalive_time = "5m"
          max_keepalive_time = "8h"
        }
      }
